pkg/helm/actions/template_test.go

```diff
@@ -0,0 +1,60 @@
+package actions
+
+import (
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chartutil"
+	kubefake "helm.sh/helm/v3/pkg/kube/fake"
+	"helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+func TestRenderManifests(t *testing.T) {
+	chart := "../testdata/influxdb-3.0.2.tgz"
+	store := storage.Init(driver.NewMemory())
+
+	actionConfig := &action.Configuration{
+		Releases:     store,
+		KubeClient:   &kubefake.PrintingKubeClient{Out: ioutil.Discard},
+		Capabilities: chartutil.DefaultCapabilities,
+		Log:          func(format string, v ...interface{}) {},
+	}
+
+	act := NewHelmActions(actionConfig)
+	rel, err := act.RenderManifests("test", chart, nil)
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.True(t, strings.HasPrefix(rel, "---"))
+}
+
+func TestRenderManifestsWithValues(t *testing.T) {
+	chart := "../testdata/influxdb-3.0.2.tgz"
+
+	store := storage.Init(driver.NewMemory())
+	actionConfig := &action.Configuration{
+		Releases:     store,
+		KubeClient:   &kubefake.PrintingKubeClient{Out: ioutil.Discard},
+		Capabilities: chartutil.DefaultCapabilities,
+		Log:          func(format string, v ...interface{}) {},
+	}
+
+	act := NewHelmActions(actionConfig)
+	js := []byte("{\"service\": { \"type\": \"NodePort\" }, \"persistence\": {\"size\": \"16Gi\"}}")
+	var m map[string]interface{}
+	json.Unmarshal(js, &m)
+
+	txt, err := act.RenderManifests("test", chart, m)
+	if err != nil {
+		t.Error(err)
+	}
+
+	assert.True(t,  strings.HasPrefix(txt, "---"))
```

Comment:
see my comment above.

Reply:
removing this assertion since I'm matching entire template string.